### PR TITLE
Issue104 EnrollmentOffering

### LINKED ISSUE #104: Split current enrolment entity into enrolment and offering entities
Offering - Year, Semester, Course [FK], Instructor [FK] (Application User)
Enrolment - Offering [FK], Student [FK] (Application User)

### PR DESCRIPTION
**Pull request by Tblack10**
Split Enrollment into Offering and Enrollment. Added Controllers for both. Newton Reference Loop Handling added to "Startup.cs". Can modify json returned from endpoints if needed.

### CODE REVIEW

File: ASPectLibrary/ApplicationUser.cs [hunk 1]
```diff
@@ -11,5 +11,6 @@ public class ApplicationUser : IdentityUser {
 
         public ICollection<Enrollment> Enrollments { get; set; }
         public IList<Membership> Memberships { get; set; }
+        public IList<Offering> Offerings { get; set; }
```
Comment (yogiduzit) [inline ASPectLibrary/ApplicationUser.cs:14]: Offerings array is still lazy loaded, can you eager load this?

Comment (yogiduzit) [reply ASPectLibrary/ApplicationUser.cs:14]: image

Comment (Tblack10) [reply ASPectLibrary/ApplicationUser.cs:14]: Offerings shouldn't be null anymore if an instructor does have an Offering.

**Review by yogiduzit — APPROVED** [review]
Looks good!

**Pull request merged**